--- a/node_modules/typescript/lib/tsserver.js	2018-01-25 16:56:27.777975000 +0400
+++ b/node_modules/typescript/lib/tsserver.js	2018-01-25 16:57:17.356049000 +0400
@@ -69576,6 +69576,7 @@
                 }
                 case "darwin":
                 case "linux":
+                case "freebsd":
                 case "android": {
                     var cacheLocation = getNonWindowsCacheLocation(process.platform === "darwin");
                     return ts.combinePaths(cacheLocation, "typescript");
--- a/extensions/typescript/node_modules/typescript/lib/tsserver.js	2018-01-25 16:56:49.937150000 +0400
+++ b/extensions/typescript/node_modules/typescript/lib/tsserver.js	2018-01-25 16:58:01.697451000 +0400
@@ -74220,6 +74220,7 @@
                 }
                 case "darwin":
                 case "linux":
+                case "freebsd":
                 case "android": {
                     var cacheLocation = getNonWindowsCacheLocation(process.platform === "darwin");
                     return ts.combinePaths(cacheLocation, "typescript");
--- a/extensions/typescript/tsconfig.json	2018-01-25 17:29:05.834842000 +0400
+++ b/extensions/typescript/tsconfig.json	2018-01-25 17:29:21.053210000 +0400
@@ -7,9 +7,9 @@
 			"es2015.promise"
 		],
 		"outDir": "./out",
-		"strictNullChecks": true,
+		"strictNullChecks": false,
 		"noImplicitAny": true,
-		"noImplicitReturns": true,
+		"noImplicitReturns": false,
 		"noUnusedLocals": true,
 		"noUnusedParameters": true
 	},
@@ -19,4 +19,4 @@
 		"out",
 		"test/colorize-fixtures"
 	]
-}
\ No newline at end of file
+}
