diff --git a/product.json b/product.json
index 8b4de36..7c4d14e 100644
--- a/product.json
+++ b/product.json
@@ -12,5 +12,67 @@
 	"win32AppUserModelId": "Microsoft.CodeOSS",
 	"darwinBundleIdentifier": "com.visualstudio.code.oss",
 	"reportIssueUrl": "https://github.com/Microsoft/vscode/issues/new",
-	"urlProtocol": "code-oss"
+	"urlProtocol": "code-oss",
+	"win32SetupExeBasename": "VSCodeSetup",
+	"downloadUrl": "https://code.visualstudio.com",
+	"welcomePage": "https://go.microsoft.com/fwlink/?LinkId=723048",
+	"updateUrl": "https://vscode-update.azurewebsites.net",
+	"extensionsGallery": {
+		"serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
+		"cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
+		"itemUrl": "https://marketplace.visualstudio.com/items"
+	},
+	"extensionTips": {
+		"ms-vscode.csharp": "{**/*.cs,**/project.json,**/global.json,**/*.csproj,**/*.sln}",
+		"ms-vscode.mono-debug": "{**/*.cs,**/*.csproj,**/*.sln}",
+		"msjsdiag.debugger-for-chrome": "{**/*.ts,**/*.tsx**/*.js,**/*.jsx,**/*.es6}",
+		"lukehoban.Go": "**/*.go",
+		"ms-vscode.PowerShell": "{**/*.ps,**/*.ps1}",
+		"Ionide.Ionide-fsharp": "{**/*.fsx,**/*.fsi,**/*.fs,**/*.ml,**/*.mli}",
+		"dbaeumer.vscode-eslint": "{**/*.js,**/*.jsx,**/*.es6}",
+		"eg2.tslint": "{**/*.ts,**/*.tsx}",
+		"Kasik96.format-indent": "{**/*.php}",
+		"felixfbecker.php-debug": "{**/*.php}",
+		"linyang95.phpmd": "{**/*.php}",
+		"ikappas.phpcs": "{**/*.php}",
+		"HvyIndustries.crane": "{**/*.php}",
+		"tht13.python": "{**/*.py}",
+		"donjayamanne.python": "{**/*.py}",
+		"magicstack.MagicPython": "{**/*.py}",
+		"saviorisdead.RustyCode": "{**/*.rs,**/*.rslib}",
+		"austin.code-gnu-global": "{**/*.c,**/*.cpp,**/*.h}",
+		"hars.CppSnippets": "{**/*.cpp,**/*.cc,**/*.h}",
+		"mitaki28.vscode-clang": "{**/*.cpp,**/*.cc,**/*.h}",
+		"ms-vscode.cpptools": "{**/*.c,**/*.cpp,**/*.cc,**/.cxx,**/*.hh,**/*.hpp,**/*.hxx,**/*.h}",
+		"seanmcbreen.Spell": "{**/*.md}",
+		"cssho.vscode-svgviewer": "{**/*.svg}",
+		"ms-vscode.latex": "{**/*.tex}",
+		"mathiasfrohlich.latexcompile": "{**/*.tex}",
+		"EditorConfig.EditorConfig": "{**/.editorconfig}",
+		"donjayamanne.githistory": "{**/.gitignore}",
+		"Unity.unity-debug": "{**/*.meta}",
+		"donjayamanne.javadebugger": "{**/*.java}",
+		"rebornix.Ruby": "{**/*.rb}",
+		"hoovercj.ruby-linter": "{**/*.rb}",
+		"misogi.ruby-rubocop": "{**/*.rb}",
+		"CraigMaslowski.erb": "{**/*.erb}",
+		"groksrc.haml": "{**/*.haml}"
+	},
+	"extensionImportantTips": {
+		"ms-vscode.csharp": "**/*.cs",
+		"lukehoban.Go": "**/*.go",
+		"donjayamanne.python": "{**/*.py}",
+		"ms-vscode.PowerShell": "{**/*.ps,**/*.ps1}",
+		"ms-vscode.cpptools": "{**/*.c,**/*.cpp,**/*.cc,**/.cxx,**/*.hh,**/*.hpp,**/*.hxx,**/*.h}"
+	},
+	"sendASmile": {
+		"reportIssueUrl": "https://go.microsoft.com/fwlink/?LinkId=534872",
+		"requestFeatureUrl": "https://go.microsoft.com/fwlink/?LinkID=533482"
+	},
+	"documentationUrl": "https://go.microsoft.com/fwlink/?LinkID=533484#vscode",
+	"releaseNotesUrl": "https://go.microsoft.com/fwlink/?LinkID=533483#vscode",
+	"twitterUrl": "https://go.microsoft.com/fwlink/?LinkID=533687",
+	"requestFeatureUrl": "https://go.microsoft.com/fwlink/?LinkID=533482",
+	"privacyStatementUrl": "https://go.microsoft.com/fwlink/?LinkId=786907",
+	"npsSurveyUrl": "https://www.research.net/r/Z2RMYRJ"
 }
\ No newline at end of file
diff --git a/scripts/npm.sh b/scripts/npm.sh
index 9d95978..e1a8abf 100755
--- a/scripts/npm.sh
+++ b/scripts/npm.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 if [[ "$OSTYPE" == "darwin"* ]]; then
 	realpath() { [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"; }
--- a/scripts/code.sh	2017-11-29 17:04:52.675340000 +0400
+++ b/scripts/code.sh	2017-12-03 09:00:02.534799000 +0400
@@ -15,7 +15,8 @@
 		CODE="./.build/electron/$NAME.app/Contents/MacOS/Electron"
 	else
 		NAME=`node -p "require('./product.json').applicationName"`
-		CODE=".build/electron/$NAME"
+		#CODE=".build/electron/$NAME"
+		CODE="../electron/dist/electron"
 	fi
 
 	INTENDED_VERSION="v`node -p "require('./package.json').electronVersion"`"
@@ -25,10 +26,12 @@
 	test -d node_modules || ./scripts/npm.sh install
 
 	# Get electron
-	(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
+	#(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
 
 	# Build
 	test -d out || ./node_modules/.bin/gulp compile
+	
+	export LD_LIBRARY_PATH=$(dirname "$(realpath "$CODE")")
 
 	# Configuration
 	export NODE_ENV=development
--- a/scripts/code-cli.sh	2017-11-29 17:04:52.675115000 +0400
+++ b/scripts/code-cli.sh	2017-12-03 09:00:21.574199000 +0400
@@ -15,7 +15,7 @@
 		CODE="./.build/electron/$NAME.app/Contents/MacOS/Electron"
 	else
 		NAME=`node -p "require('./product.json').applicationName"`
-		CODE=".build/electron/$NAME"
+		CODE="../electron/dist/electron"
 	fi
 
 	INTENDED_VERSION="v`node -p "require('./package.json').electronVersion"`"
@@ -25,11 +25,13 @@
 	test -d node_modules || ./scripts/npm.sh install
 
 	# Get electron
-	(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
+	#(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
 
 	# Build
 	test -d out || ./node_modules/.bin/gulp compile
 
+	export LD_LIBRARY_PATH=$(dirname "$(realpath "$CODE")")
+	
 	ELECTRON_RUN_AS_NODE=1 \
 	NODE_ENV=development \
 	VSCODE_DEV=1 \
--- a/scripts/test.sh	2017-11-29 17:04:52.676515000 +0400
+++ b/scripts/test.sh	2017-12-03 09:00:38.291479000 +0400
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 
 if [[ "$OSTYPE" == "darwin"* ]]; then
@@ -15,7 +15,8 @@
 	CODE="./.build/electron/$NAME.app/Contents/MacOS/Electron"
 else
 	NAME=`node -p "require('./product.json').applicationName"`
-	CODE=".build/electron/$NAME"
+	#CODE=".build/electron/$NAME"
+	CODE="../electron/dist/electron"
 fi
 
 INTENDED_VERSION="v`node -p "require('./package.json').electronVersion"`"
@@ -25,7 +26,9 @@
 test -d node_modules || ./scripts/npm.sh install
 
 # Get electron
-(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
+#(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
+
+export LD_LIBRARY_PATH=$(dirname "$(realpath "$CODE")")
 
 # Unit Tests
 export ELECTRON_ENABLE_LOGGING=1
diff --git a/scripts/test-integration.sh b/scripts/test-integration.sh
index fc7b94b..690bcc1 100755
--- a/scripts/test-integration.sh
+++ b/scripts/test-integration.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 set -e
 
 if [[ "$OSTYPE" == "darwin"* ]]; then
--- a/scripts/test-mocha.sh	2017-12-04 12:14:38.161847000 +0400
+++ b/scripts/test-mocha.sh	2017-12-04 12:15:17.872028000 +0400
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 if [[ "$OSTYPE" == "darwin"* ]]; then
 	realpath() { [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"; }
@@ -14,7 +14,8 @@
 	CODE="./.build/electron/$NAME.app/Contents/MacOS/Electron"
 else
 	NAME=`node -p "require('./product.json').applicationName"`
-	CODE=".build/electron/$NAME"
+	#CODE=".build/electron/$NAME"
+	CODE="../electron/dist/electron"
 fi
 
 INTENDED_VERSION="v`node -p "require('./package.json').electronVersion"`"
@@ -24,10 +25,11 @@
 test -d node_modules || ./scripts/npm.sh install
 
 # Get electron
-(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
+#(test -f "$CODE" && [ $INTENDED_VERSION == $INSTALLED_VERSION ]) || ./node_modules/.bin/gulp electron
 
 # Build
 test -d out || ./node_modules/.bin/gulp compile
+export LD_LIBRARY_PATH=$(dirname "$(realpath "$CODE")")
 
 # Unit Tests
 export VSCODE_DEV=1
diff --git a/src/vs/base/common/platform.ts b/src/vs/base/common/platform.ts
index 3075beb..97e6957 100644
--- a/src/vs/base/common/platform.ts
+++ b/src/vs/base/common/platform.ts
@@ -47,6 +47,7 @@ if (typeof process === 'object') {
 	_isWindows = (process.platform === 'win32');
 	_isMacintosh = (process.platform === 'darwin');
 	_isLinux = (process.platform === 'linux');
+	_isLinux = (process.platform === 'freebsd');
 	_isRootUser = !_isWindows && (process.getuid() === 0);
 	let rawNlsConfig = process.env['VSCODE_NLS_CONFIG'];
 	if (rawNlsConfig) {

--- a/src/paths.js	2017-11-28 17:13:30.169778000 +0400
+++ b/src/paths.js	2017-11-28 17:13:44.777316000 +0400
@@ -12,6 +12,7 @@
 		case 'win32': return process.env['VSCODE_APPDATA'] || process.env['APPDATA'] || path.join(process.env['USERPROFILE'], 'AppData', 'Roaming');
 		case 'darwin': return process.env['VSCODE_APPDATA'] || path.join(os.homedir(), 'Library', 'Application Support');
 		case 'linux': return process.env['VSCODE_APPDATA'] || process.env['XDG_CONFIG_HOME'] || path.join(os.homedir(), '.config');
+		case 'freebsd': return process.env['VSCODE_APPDATA'] || process.env['XDG_CONFIG_HOME'] || path.join(os.homedir(), '.config');
 		default: throw new Error('Platform not supported');
 	}
 }
--- a/src/vs/workbench/parts/codeEditor/electron-browser/selectionClipboard.ts	2017-11-28 17:15:37.043728000 +0400
+++ b/src/vs/workbench/parts/codeEditor/electron-browser/selectionClipboard.ts	2017-11-28 17:16:38.762306000 +0400
@@ -25,7 +25,7 @@
 	constructor(editor: ICodeEditor, @IContextKeyService contextKeyService: IContextKeyService) {
 		super();
 
-		if (platform.isLinux) {
+		if (0) {
 			let isEnabled = editor.getConfiguration().contribInfo.selectionClipboard;
 
 			this._register(editor.onDidChangeConfiguration((e: IConfigurationChangedEvent) => {
--- a/package.json	2017-11-28 17:27:02.422509000 +0400
+++ b/package.json	2017-11-28 17:27:13.208293000 +0400
@@ -62,7 +62,6 @@
     "cson-parser": "^1.3.3",
     "debounce": "^1.0.0",
     "documentdb": "^1.5.1",
-    "electron-mksnapshot": "~1.7.0",
     "eslint": "^3.4.0",
     "event-stream": "^3.1.7",
     "express": "^4.13.1",
--- a/build/npm/postinstall.js	2017-11-28 17:35:08.360045000 +0400
+++ b/build/npm/postinstall.js	2017-11-28 17:35:19.777141000 +0400
@@ -58,5 +58,5 @@
 }
 
 npmInstall(`build`); // node modules required for build
-npmInstall('test/smoke'); // node modules required for smoketest
+//npmInstall('test/smoke'); // node modules required for smoketest
 npmInstallBuildDependencies(); // node modules for watching, specific to host node version, not electron
