--- a/node_modules/nslog/binding.gyp	2017-12-08 22:07:06.169379000 +0400
+++ b/node_modules/nslog/binding.gyp	2017-12-08 22:07:21.419210000 +0400
@@ -37,6 +37,11 @@
             'src/nslog_linux.cc',
           ],
         }],
+        ['OS=="freebsd"', {
+          'sources': [
+            'src/nslog_linux.cc',
+          ],
+        }],
       ],
     }
   ]
--- a/node_modules/keyboard-layout/binding.gyp	2017-12-08 22:10:37.073952000 +0400
+++ b/node_modules/keyboard-layout/binding.gyp	2017-12-08 22:31:06.804238000 +0400
@@ -49,6 +49,19 @@
             ]
           }
         }],  # OS=="linux"
+        ['OS=="freebsd"', {
+          "sources": [
+            "src/keyboard-layout-manager-linux.cc",
+          ],
+          "include_dirs": [ 
+            "/usr/local/include" 
+          ],
+          "link_settings": {
+            "libraries": [
+              "-L/usr/local/lib", "-lX11", "-lxkbfile"
+            ]
+          }
+        }],  # OS=="linux"
       ]
     }
   ]
--- a/node_modules/keyboard-layout/src/keyboard-layout-manager.h	2017-12-08 22:30:01.335245000 +0400
+++ b/node_modules/keyboard-layout/src/keyboard-layout-manager.h	2017-12-08 22:30:44.572232000 +0400
@@ -3,7 +3,7 @@
 
 #include "nan.h"
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__FreeBSD__)
 #include <X11/Xlib.h>
 #endif // __linux__
 
@@ -21,7 +21,7 @@
   static NAN_METHOD(GetInstalledKeyboardLanguages);
   static NAN_METHOD(GetCurrentKeymap);
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__FreeBSD__)
   Display *xDisplay;
   XIC xInputContext;
   XIM xInputMethod;
--- a/node_modules/scrollbar-style/binding.gyp	2017-12-08 22:35:14.900905000 +0400
+++ b/node_modules/scrollbar-style/binding.gyp	2017-12-08 22:35:40.826235000 +0400
@@ -38,6 +38,11 @@
             "src/scrollbar-style-observer-non-mac.cc",
           ],
         }],  # OS=="linux"
+        ['OS=="freebsd"', {
+          "sources": [
+            "src/scrollbar-style-observer-non-mac.cc",
+          ],
+        }],  # OS=="freebsd"
       ]
     }
   ]
diff --git a/script/lib/check-chromedriver-version.js b/script/lib/check-chromedriver-version.js
index 1659f093c..851f3327e 100644
--- a/script/lib/check-chromedriver-version.js
+++ b/script/lib/check-chromedriver-version.js
@@ -10,24 +10,5 @@ module.exports = function () {
   const mksnapshotVer = buildMetadata.dependencies['electron-mksnapshot']
 
   // Always use tilde on electron-chromedriver so that it can pick up the best patch version
-  if (!chromedriverVer.startsWith('~')) {
-    throw new Error(`electron-chromedriver version in script/package.json should start with a tilde to match latest patch version.`)
-  }
-
-  if (!mksnapshotVer.startsWith('~')) {
-    throw new Error(`electron-mksnapshot version in script/package.json should start with a tilde to match latest patch version.`)
-  }
-
-  const electronVer = CONFIG.appMetadata.electronVersion
-  if (!semver.satisfies(electronVer, chromedriverVer)) {
-    throw new Error(`electron-chromedriver ${chromedriverVer} incompatible with electron ${electronVer}.\n` +
-                    'Did you upgrade electron in package.json and forget to upgrade electron-chromedriver in ' +
-                    `script/package.json to '~${semver.major(electronVer)}.${semver.minor(electronVer)}' ?`)
-  }
-
-  if (!semver.satisfies(electronVer, mksnapshotVer)) {
-    throw new Error(`electron-mksnapshot ${mksnapshotVer} incompatible with electron ${electronVer}.\n` +
-                    'Did you upgrade electron in package.json and forget to upgrade electron-mksnapshot in ' +
-                    `script/package.json to '~${semver.major(electronVer)}.${semver.minor(electronVer)}' ?`)
-  }
+ 
 }
diff --git a/script/lib/dump-symbols.js b/script/lib/dump-symbols.js
index 6b0b10bef..4ed57f6d7 100644
--- a/script/lib/dump-symbols.js
+++ b/script/lib/dump-symbols.js
@@ -6,7 +6,7 @@ const path = require('path')
 
 const CONFIG = require('../config')
 module.exports = function () {
-  if (process.platform === 'win32') {
+  if (process.platform === 'win32' || process.platform === 'freebsd') {
     console.log('Skipping symbol dumping because minidump is not supported on Windows'.gray)
     return Promise.resolve()
   } else {
--- a/script/node_modules/electron-packager/index.js	2017-12-09 15:14:33.273907000 +0400
+++ b/script/node_modules/electron-packager/index.js	2017-12-09 12:57:29.690386000 +0400
@@ -19,6 +19,7 @@
   // Maps to module ID for each platform (lazy-required if used)
   darwin: './mac',
   linux: './linux',
+  freebsd: './linux',
   mas: './mac', // map to darwin
   win32: './win32'
 }
diff --git a/src/atom-paths.js b/src/atom-paths.js
index 39a768e91..28b479604 100644
--- a/src/atom-paths.js
+++ b/src/atom-paths.js
@@ -19,6 +19,7 @@ const getAppDirectory = () => {
     case 'darwin':
       return process.execPath.substring(0, process.execPath.indexOf('.app') + 4)
     case 'linux':
+    case 'freebsd':
     case 'win32':
       return path.join(process.execPath, '..')
   }
diff --git a/src/crash-reporter-start.js b/src/crash-reporter-start.js
index 035a94e3d..3e2716270 100644
--- a/src/crash-reporter-start.js
+++ b/src/crash-reporter-start.js
@@ -1,10 +1,3 @@
 module.exports = function (extra) {
-  const {crashReporter} = require('electron')
-  crashReporter.start({
-    productName: 'Atom',
-    companyName: 'GitHub',
-    submitURL: 'https://crashreporter.atom.io',
-    uploadToServer: false,
-    extra: extra
-  })
+  
 }
diff --git a/src/update-process-env.js b/src/update-process-env.js
index 00bb13927..e8ae4ae99 100644
--- a/src/update-process-env.js
+++ b/src/update-process-env.js
@@ -12,7 +12,8 @@ const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set([
 
 const PLATFORMS_KNOWN_TO_WORK = new Set([
   'darwin',
-  'linux'
+  'linux',
+  'freebsd'
 ])
 
 async function updateProcessEnv (launchEnv) {
--- a/node_modules/dugite/build/lib/git-environment.js	2018-06-04 22:02:09.259250000 +0400
+++ b/node_modules/dugite/build/lib/git-environment.js	2018-06-04 22:02:21.512639000 +0400
@@ -17,6 +17,9 @@
     if (process.env.LOCAL_GIT_DIRECTORY) {
         return path.resolve(process.env.LOCAL_GIT_DIRECTORY);
     }
+    else if (process.platform === 'freebsd') {
+        return '/usr/local'
+    }
     else {
         return resolveEmbeddedGitDir();
     }
@@ -28,6 +31,7 @@
     const gitDir = resolveGitDir();
     if (process.platform === 'darwin' ||
         process.platform === 'linux' ||
+        process.platform === 'freebsd' ||
         process.platform === 'android') {
         return path.join(gitDir, 'bin', 'git');
     }
@@ -49,6 +53,7 @@
     const gitDir = resolveGitDir();
     if (process.platform === 'darwin' ||
         process.platform === 'linux' ||
+        process.platform === 'freebsd' ||
         process.platform === 'android') {
         return path.join(gitDir, 'libexec', 'git-core');
     }
@@ -86,7 +91,7 @@
             delete env.Path;
         }
     }
-    if (process.platform === 'darwin' || process.platform === 'linux') {
+    if (process.platform === 'darwin' || process.platform === 'linux' || process.platform === 'freebsd') {
         // templates are used to populate your .git folder
         // when a repository is initialized locally
         const templateDir = `${gitDir}/share/git-core/templates`;
diff --git a/src/config-schema.js b/src/config-schema.js
index 2ff68be86..cbfd1c1bf 100644
--- a/src/config-schema.js
+++ b/src/config-schema.js
@@ -530,7 +530,7 @@ const configSchema = {
   }
 }
 
-if (['win32', 'linux'].includes(process.platform)) {
+if (['win32', 'linux', 'freebsd'].includes(process.platform)) {
   configSchema.core.properties.autoHideMenuBar = {
     type: 'boolean',
     default: false,
diff --git a/src/main-process/atom-application.coffee b/src/main-process/atom-application.coffee
index f6802705e..c1d278878 100644
--- a/src/main-process/atom-application.coffee
+++ b/src/main-process/atom-application.coffee
@@ -165,7 +165,7 @@ class AtomApplication
     @windowStack.removeWindow(window)
     if @getAllWindows().length is 0
       @applicationMenu?.enableWindowSpecificItems(false)
-      if process.platform in ['win32', 'linux']
+      if process.platform in ['win32', 'linux', 'freebsd']
         app.quit()
         return
     @saveState(true) unless window.isSpec
diff --git a/src/main-process/atom-window.coffee b/src/main-process/atom-window.coffee
index ca3995c05..d11b232c0 100644
--- a/src/main-process/atom-window.coffee
+++ b/src/main-process/atom-window.coffee
@@ -40,7 +40,7 @@ class AtomWindow
 
     # Don't set icon on Windows so the exe's ico will be used as window and
     # taskbar's icon. See https://github.com/atom/atom/issues/4811 for more.
-    if process.platform is 'linux'
+    if process.platform is 'linux' or process.platform is 'freebsd'
       options.icon = @constructor.iconPath
 
     if @shouldAddCustomTitleBar()
diff --git a/src/main-process/auto-update-manager.coffee b/src/main-process/auto-update-manager.coffee
index 0e4144c1a..c627e8604 100644
--- a/src/main-process/auto-update-manager.coffee
+++ b/src/main-process/auto-update-manager.coffee
@@ -68,6 +68,8 @@ class AutoUpdateManager
         @setState(UnsupportedState) unless autoUpdater.supportsUpdates()
       when 'linux'
         @setState(UnsupportedState)
+      when 'freebsd'
+        @setState(UnsupportedState)
 
   emitUpdateAvailableEvent: ->
     return unless @releaseVersion?
diff --git a/src/safe-clipboard.coffee b/src/safe-clipboard.coffee
index 1f91803e2..fb6cec5c6 100644
--- a/src/safe-clipboard.coffee
+++ b/src/safe-clipboard.coffee
@@ -1,6 +1,6 @@
 # Using clipboard in renderer process is not safe on Linux.
 module.exports =
-  if process.platform is 'linux' and process.type is 'renderer'
+  if (process.platform is 'linux' or process.platform is 'freebsd') and process.type is 'renderer'
     require('electron').remote.clipboard
   else
     require('electron').clipboard
diff --git a/src/text-editor-component.js b/src/text-editor-component.js
index 91ea18361..3c491321d 100644
--- a/src/text-editor-component.js
+++ b/src/text-editor-component.js
@@ -1596,7 +1596,7 @@ class TextEditorComponent {
     // the paste event. And since we don't use the `paste` event for any
     // behavior in Atom, we can no-op the event to eliminate this issue.
     // See https://github.com/atom/atom/pull/15183#issue-248432413.
-    if (this.getPlatform() === 'linux') event.preventDefault()
+    if (this.getPlatform() === 'linux' || his.getPlatform() === 'freebsd') event.preventDefault()
   }
 
   didTextInput (event) {
@@ -1759,7 +1759,7 @@ class TextEditorComponent {
     // On Linux, position the cursor on middle mouse button click. A
     // textInput event with the contents of the selection clipboard will be
     // dispatched by the browser automatically on mouseup.
-    if (platform === 'linux' && button === 1) {
+    if ((platform === 'linux' || platform === 'freebsd') && button === 1) {
       const selection = clipboard.readText('selection')
       const screenPosition = this.screenPositionForMouseEvent(event)
       model.setCursorScreenPosition(screenPosition, {autoscroll: false})
@@ -2477,7 +2477,7 @@ class TextEditorComponent {
   didChangeSelectionRange () {
     const {model} = this.props
 
-    if (this.getPlatform() === 'linux') {
+    if (this.getPlatform() === 'linux' || this.getPlatform() === 'freebsd') {
       if (this.selectionClipboardImmediateId) {
         clearImmediate(this.selectionClipboardImmediateId)
       }
diff --git a/src/window-event-handler.js b/src/window-event-handler.js
index 6d380819b..7fb49c39e 100644
--- a/src/window-event-handler.js
+++ b/src/window-event-handler.js
@@ -37,7 +37,7 @@ class WindowEventHandler {
       'window:toggle-dev-tools': this.handleWindowToggleDevTools
     }))
 
-    if (['win32', 'linux'].includes(process.platform)) {
+    if (['win32', 'linux', 'freebsd'].includes(process.platform)) {
       this.subscriptions.add(this.atomEnvironment.commands.add(this.window,
         {'window:toggle-menu-bar': this.handleWindowToggleMenuBar})
       )

 


