diff --git a/native_mate/converter.cc b/native_mate/converter.cc
index 97fa455..9edf940 100644
--- a/native_mate/converter.cc
+++ b/native_mate/converter.cc
@@ -32,7 +32,7 @@ bool Converter<bool>::FromV8(Isolate* isolate, Local<Value> val, bool* out) {
   return true;
 }
 
-#if !defined(OS_LINUX)
+#if !defined(OS_LINUX) && !defined(OS_FREEBSD)
 Local<Value> Converter<unsigned long>::ToV8(Isolate* isolate,
                                              unsigned long val) {
   return MATE_INTEGER_NEW(isolate, val);
diff --git a/native_mate/converter.h b/native_mate/converter.h
index ebb6273..d5606e4 100644
--- a/native_mate/converter.h
+++ b/native_mate/converter.h
@@ -49,7 +49,7 @@ struct Converter<bool> {
                      bool* out);
 };
 
-#if !defined(OS_LINUX)
+#if !defined(OS_LINUX) && !defined(OS_FREEBSD)
 template<>
 struct Converter<unsigned long> {
   static v8::Local<v8::Value> ToV8(v8::Isolate* isolate,
